contributors/devel/sig-architecture/api_changes.md

```diff
@@ -128,6 +128,9 @@ continue to function as they did previously, even when your change is in use.
 does not include your change and have no impact on API objects which do not use
 your change.  API objects that use your change will be impacted in case of a
 rollback.
+7. Reading an unmodified object via the same API version, across different
```

Comment:
I think this is an impossible requirement given that we have webhooks which can do arbitrary things to objects.

Reply:
I think this is saying two reads of the same apiversion/resource at the same resourceVersion at different points in time must return the same data

Reply:
Wasn't this the intention of the metadata.generation field? To allow understanding if spec had been changed from a point in the past?

Reply:
I said in a different reply - defaults set on CREATE are not the issue.  It's defaults which are set on GET.  I'm not saying that I should GET back exactly what I CREATEd, but that any 2 GETs should return the same document, as long as the object has not actually been modified.

Reply:
Why is that an important property? Are you looking to guarantee byte-for-byte identical? In proto, json, both?

Reply:
I am looking for a way to allow changes to structs that fall under PodSpec without triggering a re-deploy each time.  This PR is a proposal of one approach to it.  So I only think it is important in that context.

Reply:
Why is fixing the controllers not an option?

Reply:
If we have a good fix, yes please.  I forgot about generation - is that something that can apply fairly universally